flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -1560,6 +1561,13 @@ class CodeGenerator(
         requireArray(array)
         generateArrayElement(this, array)
 
+      case ScalarSqlFunctions.CONCAT | ScalarSqlFunctions.CONCAT_WS =>
+        this.config.setNullCheck(false)
```

Comment:
We cannot modify the config here. Maybe it makes sense not to use a `CallGenerator` but add the logic to `ScalarOperators`. Other arrays functions such as `generateArrayCardinality`, `generateArrayElement` are there as well.